doc/encoding.rst

```diff
@@ -10,27 +10,59 @@ Encoding or vectorizing creates numerical features from the data,
 converting dataframes, strings, dates... Different encoders are suited
 for different types of data.
 
+.. _table_vectorizer:
+
 Turning a dataframe into a numerical feature matrix
-----------------------------------------------------
+---------------------------------------------------
 
 A dataframe can comprise columns of all kind of types. A good numerical
 representation of these columns help analytics and statistical learning.
 
 The :class:`TableVectorizer` gives a turn-key solution by applying
-different data-specific encoder to the different columns. It makes
-heuristic choices that are not necessarily optimal but is typically a
-very good baseline.
-
-The function :func:`tabular_learner` is an easy way to create a machine-learning
-model that works well on tabular data. It is given a scikit-learn estimator and
-creates a :class:`~sklearn.pipeline.Pipeline` combining a
-:class:`TableVectorizer`, optional missing value imputation and the provided
-estimator.
+different data-specific encoder to the different columns. It makes reasonable
+heuristic choices that are not necessarily optimal since it is not aware of the learner
+used for the machine learning task). However, it already provides a typically very good
+baseline.
+
+The function :func:`tabular_learner` goes the extra mile by creating a machine-learning
+model that works well on tabular data. This model combines a :class:`TableVectorizer`
+with a provided scikit-learn estimator. Depending whether or not the final estimator
+natively support missing values, a missing value imputer step is added before the
+final estimator. The parameters of the :class:`TableVectorizer` are chosen based on the
+type of the final estimator.
+
+.. list-table:: Parameter values choice of :class:`TableVectorizer` when using :func:`tabular_learner` function
```

Comment:
This table should look like:

![image](https://github.com/skrub-data/skrub/assets/7454015/f5ad2909-e9e3-4fed-b12f-b7544337f412)

Reply:
that's a good idea, super useful! the non-hist gradient boosting would go in the same column as randomforest but just "RandomForest" is good it's better to keep the headings short